wallet/src/key_chain/account_key_chain/mod.rs

```diff
@@ -173,17 +183,44 @@ impl AccountKeyChain {
         }
     }
 
+    pub fn get_private_key_for_destination(
+        &self,
+        destination: &Destination,
+    ) -> KeyChainResult<ExtendedPrivateKey> {
```

Comment:
Can we return `KeyChainResult<Option<ExtendedPrivateKey>>` to distinguish between errors and not finding the key?